augur/frequency_estimators.py

```diff
@@ -54,18 +59,29 @@ def get_pivots(observations, pivot_interval, start_date=None, end_date=None, piv
     pivot_end = end_date if end_date else np.ceil(np.max(observations) / pivot_frequency) * pivot_frequency
 
     if pivot_interval_units == "months":
-        offset = "%sMS" % pivot_interval
+        delta_keywords = {"months": pivot_interval}
     elif pivot_interval_units == "weeks":
-        offset = "%sW" % pivot_interval
+        delta_keywords = {"weeks": pivot_interval}
     else:
         raise ValueError(f"The given interval unit '{pivot_interval_units}' is not supported.")
 
-    datetime_pivots = pd.date_range(
-        float_to_datestring(pivot_start),
-        float_to_datestring(pivot_end),
-        freq = offset
-    )
-    pivots = np.array([timestamp_to_float(pivot) for pivot in datetime_pivots])
+    # Construct standard Python date instances from numeric dates via ISO-8601
+    # dates and the corresponding delta time for the interval between pivots.
+    start = datetime.datetime.strptime(float_to_datestring(pivot_start), "%Y-%m-%d")
+    end = datetime.datetime.strptime(float_to_datestring(pivot_end), "%Y-%m-%d")
+    delta = relativedelta(**delta_keywords)
+
+    # Start calculating pivots from the end date (inclusive), working backwards
+    # in time by a time delta that matches the user-requested interval. Include
+    # the start date in the final pivots when the interval spacing allows that
+    # date to be included.
+    pivots = deque([])
+    pivot = end
+    while pivot >= start:
+        pivots.appendleft(pivot)
+        pivot = pivot - delta
+
+    pivots = np.array([numeric_date(pivot.strftime("%Y-%m-%d")) for pivot in pivots])
```

Comment:
There's some unnecessary round-tripping of date formats. I've created a separate PR to improve this: #1124. Once that's merged, this can be simplified:

```suggestion
    pivots = np.array([numeric_date(pivot) for pivot in pivots])
```

Reply:
Yeah, it felt bad to write, but I also wasn't ready to go down the rabbit hole of modifying `numeric_date` for this PR. Thank you for addressing this in #1124!

Reply:
Resolved in df62699.